test_utils/crates/mock-atlantic-server/src/server.rs

```diff
@@ -0,0 +1,398 @@
+                    warn!("Failed to parse declaredJobSize field: {}", e);
+                    StatusCode::BAD_REQUEST
+                })?);
+                debug!("Parsed declaredJobSize: {:?}", declared_job_size);
+            }
+            Some("cairoVersion") => {
+                cairo_version = Some(field.text().await.map_err(|e| {
+                    warn!("Failed to parse cairoVersion field: {}", e);
+                    StatusCode::BAD_REQUEST
+                })?);
+                debug!("Parsed cairoVersion: {:?}", cairo_version);
+            }
+            Some("cairoVm") => {
+                cairo_vm = Some(field.text().await.map_err(|e| {
+                    warn!("Failed to parse cairoVm field: {}", e);
+                    StatusCode::BAD_REQUEST
+                })?);
+                debug!("Parsed cairoVm: {:?}", cairo_vm);
+            }
+            Some("result") => {
+                result_type = Some(field.text().await.map_err(|e| {
+                    warn!("Failed to parse result field: {}", e);
+                    StatusCode::BAD_REQUEST
+                })?);
+                debug!("Parsed result: {:?}", result_type);
+            }
+            Some("externalId") => {
+                external_id = Some(field.text().await.map_err(|e| {
+                    warn!("Failed to parse externalId field: {}", e);
+                    StatusCode::BAD_REQUEST
+                })?);
+                debug!("Parsed externalId: {:?}", external_id);
+            }
+            Some("pieFile") | Some("inputFile") | Some("programFile") => {
+                let data = field.bytes().await.map_err(|e| {
+                    warn!("Failed to read file data: {}", e);
+                    StatusCode::BAD_REQUEST
+                })?;
+                info!("Received file data of {} bytes for field {:?}", data.len(), field_name);
+                // For mock server, we just log the file data but don't process it
+            }
+            Some(name) => {
+                // Handle any other fields by consuming them
+                let value = field.text().await.map_err(|e| {
+                    warn!("Failed to parse field {}: {}", name, e);
+                    StatusCode::BAD_REQUEST
+                })?;
+                debug!("Parsed field {}: {}", name, value);
+            }
+            None => {
+                warn!("Received field with no name");
+                let _ = field.bytes().await;
+            }
+        }
+    }
+
+    // Determine job ID based on layout
+    let job_id = match layout.as_deref() {
+        Some("recursive_with_poseidon") => "01JXMXQAX4KNNSQDKDZTSHG8FC".to_string(),
+        Some("dynamic") => "01JXMTC7TZMSNDTJ88212KTH7W".to_string(),
+        _ => {
+            // Default to dynamic layout ID for other layouts
+            debug!("Using default job ID for layout: {:?}", layout);
+            "01JXMTC7TZMSNDTJ88212KTH7W".to_string()
+        }
+    };
```

Comment:
we don't really need to check for the dynamic layout tho?
```suggestion
    let job_id = match layout.as_deref() {
        Some("recursive_with_poseidon") => "01JXMXQAX4KNNSQDKDZTSHG8FC".to_string(),
        _ => {
            // Default to dynamic layout ID for other layouts
            debug!("Using default job ID for layout: {:?}", layout);
            "01JXMTC7TZMSNDTJ88212KTH7W".to_string()
        }
    };
```